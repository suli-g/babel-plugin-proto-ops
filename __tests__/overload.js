const { transform } = require('@babel/core')
const plugin = require('../')

const tests = [
    '1 + 1',
    'x + 1',
    '{} + 1',
    '1 + {}',
    '1 + x',
    'this + 1',
    'this.valueOf() + 1',
    'this.valueOf() + x',
    'this.valueOf() + x.valueOf()',
    '{a: 1} + {b: 2}',
    'a + {b: 2}',
    `function x() {
        return this.valueOf() + x.valueOf()
    }`
]

it('works', () => {
    for (const test of tests) {
        const { code } = transform(test, { plugins: [plugin]})
        expect(code).toMatchSnapshot()
    }
})