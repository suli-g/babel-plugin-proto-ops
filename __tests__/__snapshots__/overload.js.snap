// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `"1[\\"+\\"](1);"`;

exports[`works 2`] = `"x[\\"+\\"](1);"`;

exports[`works 3`] = `
"{}
+1;"
`;

exports[`works 4`] = `"1[\\"+\\"]({});"`;

exports[`works 5`] = `"1[\\"+\\"](x);"`;

exports[`works 6`] = `"this[\\"+\\"](1);"`;

exports[`works 7`] = `"this.valueOf()[\\"+\\"](1);"`;

exports[`works 8`] = `"this.valueOf()[\\"+\\"](x);"`;

exports[`works 9`] = `"this.valueOf()[\\"+\\"](x.valueOf());"`;

exports[`works 10`] = `
"{
  a: 1;
}
+{
  b: 2
};"
`;

exports[`works 11`] = `
"a[\\"+\\"]({
  b: 2
});"
`;

exports[`works 12`] = `
"function x() {
  return this.valueOf()[\\"+\\"](x.valueOf());
}"
`;
